cleverhans/attacks.py

```diff
@@ -21,22 +21,30 @@ class Attack(object):
   """
   __metaclass__ = ABCMeta
 
-  def __init__(self, model, back='tf', sess=None, dtypestr='float32'):
+  def __init__(self, model, sess=None, dtypestr='float32', **kwargs):
     """
     :param model: An instance of the cleverhans.model.Model class.
-    :param back: The backend to use. Currently 'tf' is the only option.
-    :param sess: The tf session to run graphs in
-    """
-    if back == 'tf':
-      import tensorflow as tf
-      self.tf_dtype = tf.as_dtype(dtypestr)
-      if sess is None:
-        sess = tf.get_default_session()
-    else:
-      raise ValueError("Backend argument must be 'tf'.")
+    :param sess: The tf session to run graphs in.
+    :param dtypestr: Floating point precision to use (change to float64
+                     to avoid numerical instabilities).
+    :param back: (deprecated and will be removed on or after 2019-03-26).
+                 The backend to use. Currently 'tf' is the only option.
+    """
+    if 'back' in kwargs:
+      if kwargs['back'] == 'tf':
+        warnings.warn("Argument back to attack constructors is not needed"
+                      " anymore and will be removed on or after 2019-03-26."
+                      " All attacks are implemented using TensorFlow.")
+      else:
+        raise ValueError("Backend argument must be 'tf' and is now deprecated"
+                         "It will be removed on or after 2019-03-26.")
 
+    self.tf_dtype = tf.as_dtype(dtypestr)
     self.np_dtype = np.dtype(dtypestr)
 
```

Comment:
Add a check on the type of `sess`. This will also make sure that an old positional argument passed for `back` doesn't end up going to `sess` now that `back` has been removed.